### ISSUE TITLE
[BUG] Excel addin triggers new model/page initialization on first navigation event

### ISSUE THREAD

**Issue by kMutagene** [nfdi4plants/Swate#21]
Clicking the taskpane button initializes the addin with an initial model. When changing the page (by clicking any navigation tab) the first time, a new initialization is triggered. This should not happen, and could be either due to the addin not using a common runtime or the taskpane buttons not pointing on the root page.

**Comment by Freymaurer** [comment]
I tried multiple alternatives for routing to fix this problem. All of which were very lackluster
- non hash-based routing ->completly reload the page on each reroute
- non hash-based routing with subpage as query param of Route.Home -> same issue as above

To fix most problems i first placed most of the init cmd messages in a batch after excel initializes. This fixed the reappearing yellow warning box that there is no annotationTable even though one was created before rerouting.

```fsharp 
let init (pageOpt: Routing.Route option) : Model * Cmd<Msg> =
    let initialModel = initializeModel pageOpt
    let route = (parseHash Routing.Routing.route) Browser.Dom.document.location
    // The initial command from urlUpdate is not needed yet. As we use a reduced variant of subModels with no own Msg system.
    let model, _ = urlUpdate route initialModel
/// This is the important part. Only initializeAddIn is executed here
    let initialCmd =
        Cmd.batch [
            Cmd.OfPromise.either
                initializeAddIn
                ()
                (fun x -> (x.host.ToString(),x.platform.ToString()) |> Initialized |> ExcelInterop )
                (fun x -> x |> GenericError |> Dev)
        ]
    model, initialCmd
```
Instead the other init messages are now chained after initializeAddIn returned successful

```
    | Initialized (h,p) ->
        let welcomeMsg = sprintf "Ready to go in %s running on %s" h p

        let nextState = {
            currentState with
                Host        = h
                Platform    = p
        }
/// This is the important part. Only initializeAddIn is executed here
        let cmd =
            Cmd.batch [
                Cmd.ofMsg (FetchAllOntologies |> Request |> Api)
                Cmd.OfPromise.either
                    OfficeInterop.checkIfAnnotationTableIsPresent
                    ()
                    (AnnotationTableExists >> ExcelInterop)
                    (GenericError >> Dev)
                Cmd.ofMsg (("Info",welcomeMsg) |> (GenericLog >> Dev))
            ]

        nextState, cmd
```

After these steps the warning label shortly reappears but is then again removed, as after the excel init it finds that the table already exists.

BUT all of this can be fixed by using hash based rerouting only to navigate to a specific entrypoint to the app and then rerouting by updating the PageState in the model. This is implemented in commit e05d95a5 on my fork.

Will do a PR soon.

**Comment by Freymaurer** [comment]
merged PR to developer
